Combining issue 19 through 22

SergejMuhic opened the pull request:
https://github.com/AECgeeks/infra-repo-issue-test-1/issues/19

to

https://github.com/AECgeeks/infra-repo-issue-test-1/issues/22

Some psets are new and the facility part type enum has not been merged yet. Those psets will need some additional love.

SergejMuhic committed:
Adding IfcVoxelGrid

SergejMuhic committed:
Code review suggestion

SergejMuhic committed:
Reconciled

SergejMuhic committed:
Reconcile

SergejMuhic committed:
Adding attribute `IfcDerivedUnit.Name` and documentation

SergejMuhic committed:
Added IfcDiscreteAccessoryTypeEnum FILLER and FLASHING

SergejMuhic committed:
Implementing changes

SergejMuhic committed:
The fix

https://github.com/AECgeeks/infra-repo-issue-test-1/issues/19

to

https://github.com/AECgeeks/infra-repo-issue-test-1/issues/22

larswik committed:
Issue #424

larswik committed:
Added enumeration constants SIREN and BEACON to IfcAudioVisualApplianceTypeEnum

Fixes #423

larswik committed:
Adding JET PDT for IfcFan

Fixes #404

larswik committed:
Adding IfcEarthingElement/IfcEarthingElementType as new subtypes of IfcFlowTerminal/IfcFlowTerminalType

Fixes #403

larswik committed:
PDT changes for IfcDistributionsystemEnum

Fixes #402
Modified description for MONITORINGSYSTEM (more generic)
Added SAFETY enum constant.

larswik committed:
New PDTs for IfcBuiltSystem

larswik committed:
Add IfcFillElement/IfcFillElementType

Fixes #399

larswik committed:
Made PredefinedType optional

larswik committed:
Made PredefinedType optional

larswik committed:
Adding PDT FIREPROTECTION

Fixes #389 passive fireprotection

larswik committed:
Update FIREPROTECTION_3DXDyU9IvAxeEaZ1cVyZ_M.xml

Updated definition for FIREPROTECTION to make it more generic.
Fixes #389 - active fire protection

larswik committed:
USERDEFINED & NOTDEFINED

Reusing instead of creating new ones.

larswik committed:
USERDEFINED & NOTDEFINED

Reused existing constants instead of creating new.

larswik committed:
Update IfcMaterial documentation

Fixes #454

larswik committed:
Adding IfcObservation and IfcGeoScienceObservation

Fixes #391 (except IfcProductData, now IfcComplementaryData which was merged from a separate PR #440)

larswik committed:
Adding IfcVoxelData and subclasses

Fixes #448

larswik commented on IFC4x3/PropertyConstants/u/UNSET_1n8r4XYi950Rj6LjJ0X9h_.xml:
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="UNSET_1n8r4XYi950Rj6LjJ0X9h_" Name="UNSET" UniqueId="71235121-8ac2-4501-bb46-56d4c0849afe">
```
Is there a need for a new UNSET constant? Why not NOTDEFINED?

SergejMuhic replied:
To address with @aothms

larswik commented on IFC4x3/Sections/Domain specific data schemas/Schemas/IfcPortsAndWaterwaysDomain/PropertySets/Pset_GateHeadCommon/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_GateHeadCommon" UniqueId="8dbb748d-b6f3-4559-8587-0575f097df30" ApplicableType="IfcFacilityPart/IfcMarinePartTypeEnum(.GATEHEAD.)" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_GateHeadCommon" UniqueId="8dbb748d-b6f3-4559-8587-0575f097df30" ApplicableType="IfcFacilityPart/IfcMarinePartTypeEnum(.GATEHEAD.)" PropertySetType="PSET_OCCURRENCEDRIVEN">
```
This ApplicableType-string would probably change to "IfcMarineFacilityPart/GATEHEAD" after the redesign of facility part structure.

SergejMuhic replied:
Yes, indeed. But we are handling psets in a separate PR, so let us leave it.

larswik commented on IFC4x3/Sections/Domain specific data schemas/Schemas/IfcPortsAndWaterwaysDomain/PropertySets/Pset_MarinePartChamberCommon/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_ChamberCommon" UniqueId="63a907b1-a065-46f5-adbe-1bac26deb245" ApplicableType="IfcFacilityPart/IfcMarinePartTypeEnum(.CHAMBER.)" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_MarinePartChamberCommon" UniqueId="63a907b1-a065-46f5-adbe-1bac26deb245" ApplicableType="IfcFacilityPart/IfcMarinePartTypeEnum(.CHAMBER.)" PropertySetType="PSET_OCCURRENCEDRIVEN">
```
This ApplicableType-string would probably change to "IfcMarineFacilityPart/CHAMBER" after the redesign of facility part structure.

larswik commented on IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_RailwayLevelCrossing/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_RailwayLevelCrossing" UniqueId="7f547f91-6dd0-161c-9824-1f608dd660d9" ApplicableType="IfcFacilityPart/IfcFacilityPartCommonTypeEnum(.LEVELCROSSING.)" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_RailwayLevelCrossing" UniqueId="7f547f91-6dd0-161c-9824-1f608dd660d9" ApplicableType="IfcFacilityPart/IfcFacilityPartCommonTypeEnum(.LEVELCROSSING.)" PropertySetType="PSET_OCCURRENCEDRIVEN">
```
This ApplicableType-string would probably change after the redesign of facility part structure.

larswik commented on IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_RailwayTrackStructurePart/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_RailwayTrackStructurePart" UniqueId="011923c1-fbf2-84f9-6215-e3ea43caebc9" ApplicableType="IfcFacilityPart/IfcRailwayPartTypeEnum(.TRACKSTRUCTURE.),IfcFacilityPart/IfcRailwayPartTypeEnum(.PLAINTRACKSUPESTRUCTURE.),IfcFacilityPart/IfcRailwayPartTypeEnum(.TURNOUTSUPERSTRUCTURE.),IfcFacilityPart/IfcRailwayPartTypeEnum(. ...
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_RailwayTrackStructurePart" UniqueId="011923c1-fbf2-84f9-6215-e3ea43caebc9" ApplicableType="IfcFacilityPart/IfcRailwayPartTypeEnum(.TRACKSTRUCTURE.),IfcFacilityPart/IfcRailwayPartTypeEnum(.PLAINTRACKSUPESTRUCTURE.),IfcFacilityPart/IfcRailwayPartTypeEnum(.TURNOUTSUPERSTRUCTURE.),IfcFacilityPart/IfcRailwayPartTypeEnum(. ...
```
These ApplicableType-strings would probably change after the redesign of facility part structure.

larswik commented on IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoadDomain/PropertySets/Pset_RoadDesignCriteriaCommon/DocPropertySet.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_RoadDesignCriteriaCommon" UniqueId="3c1864c7-6e95-46c7-a52f-5edba67bfffb" ApplicableType="IfcRoad,IfcFacilityPart/IfcRoadPartTypeEnum(.ROADSEGMENT.),IfcFacilityPart/IfcFacilityPartCommonTypeEnum(.SEGMENT.),IfcFacilityPart/IfcFacilityPartCommonTypeEnum(.JUNCTION.),IfcFacilityPart/IfcFacilityPartCommonTypeEnum(.LEVELCR ...
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_RoadDesignCriteriaCommon" UniqueId="3c1864c7-6e95-46c7-a52f-5edba67bfffb" ApplicableType="IfcRoad,IfcFacilityPart/IfcRoadPartTypeEnum(.ROADSEGMENT.),IfcFacilityPart/IfcFacilityPartCommonTypeEnum(.SEGMENT.),IfcFacilityPart/IfcFacilityPartCommonTypeEnum(.JUNCTION.),IfcFacilityPart/IfcFacilityPartCommonTypeEnum(.LEVELCR ...
```
These ApplicableType-strings would probably change after the redesign of facility part structure.

larswik reviewed (commented):
A few comments regarding ApplicableType-strings and the redesign of facility part structure.

pjanck commented:
Except for Lars comments, of course :P

SergejMuhic committed:
Update IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricModelResource/Entities/IfcVoxelGrid/DocEntity.xml

anborr reviewed (approved):
Check ApplicableTypes for consitency with new IfcFacilityPartType

SergejMuhic committed:
Prepairing tunnel publication

SergejMuhic committed:
Merge pull request #406 from bSI-InfraRoom/feature/materialAndProfileDriven

Define and assign PSET MATERIALDRIVEN and PSET PROFILEDRIVEN issue 110

SergejMuhic committed:
Merge pull request #413 from bSI-InfraRoom/feature/RAILWAYTRACKandRAILWAYLINEpsetAssociation

RAILWAYTRACK and RAILWAYLINE + pset association issue 305

SergejMuhic committed:
Merge pull request #417 from bSI-InfraRoom/feature/ifcDerivedUnitName

Adding attribute `IfcDerivedUnit.Name` and enums `FILLER` and `FLASHING` to `IfcDiscreteAccessoryTypeEnum`

SergejMuhic committed:
Merge pull request #418 from bSI-InfraRoom/feature/removeStandardElementedCase

Remove elemented and standard cases issue 53

SergejMuhic committed:
Update IFC4x3/Sections/Resource definition data schemas/Schemas/IfcExternalReferenceResource/Entities/IfcDatasetInformation/DocEntity.xml

SergejMuhic committed:
Merge pull request #429 from larswik/Feature/IfcFan-PDT-for-Jet-fan

Adding JET PDT for IfcFan

SergejMuhic committed:
Apply suggestions from code review

SergejMuhic committed:
Merge pull request #456 from larswik/Change-IfcMaterial-description

Update IfcMaterial documentation

larswik committed:
Changed Ifc*VoxelValues to Ifc*VoxelData

larswik committed:
Update IFC4x3/Constants/f/FIREPROTECTION_2Z7j5jACn9tOoK92Pdiwhi.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Moved new enums above USERDEFINED

larswik committed:
Update IFC4x3/Constants/f/FIREPROTECTION_3DXDyU9IvAxeEaZ1cVyZ_M.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Moved new enum above USERDEFINED

larswik committed:
Update IFC4x3/Constants/g/GROUNDINGMESH_1WFsqjqs54qPDn6VzU$xcd.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Update IFC4x3/Constants/f/FIXEDTERMINAL_3HhDuEn2f2nP6ZL0SQeoy3.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Update IFC4x3/Constants/e/EARTHINGSTRIP_3Fh7NdPNTF3fo7Avf2LN6_.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Update IFC4x3/Constants/g/GROUNDINGPLATE_3lDZcwSjL5iAoe6FsC_G2e.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Update IFC4x3/Constants/g/GROUNDINGROD_25zpcaPd9AvhSIrZdgJ$rm.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Documentation for WRs

larswik committed:
Update IFC4x3/Constants/l/LABTESTRESULT_2XsgKLRhL7Ngp7nQbm_mzA.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Update IFC4x3/Constants/m/MAPPEDFEATURE_1NxCCT1ufFdf6DiOSSB4Om.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcObservation/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Fixed according to received comments

- Hint on how actual observation data is linked with IfcObservation
- "Container" => "Enumeration"
- Fixing documentation for IfcGeoScienceObservation
- WR => HasObjectType
- PredefinedType => Mandatory

larswik committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcKernel/Entities/IfcRelAssociatesDataset/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Updated according to comments.

SergejMuhic committed:
Merge pull request #427 from larswik/Feature/External-datasets

possibility to reference external datasets (IfcExternalDatasetReference)

SergejMuhic committed:
Apply suggestions from code review

Co-authored-by: larswik <lars.wikstrom@triona.se>

SergejMuhic committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceObservation/Documentation.md

SergejMuhic committed:
Merge pull request #457 from larswik/Feature/IfcObservation-IfcGeoScienceObservation

Adding IfcObservation and IfcGeoScienceObservation

larswik committed:
Update IFC4x3/Constants/i/INVERTFILL_0Rt828ifP9gPsg6uIL6_Mc.xml

Co-authored-by: anborr <73635642+anborr@users.noreply.github.com>

larswik committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcFillElementType/Documentation.md

Co-authored-by: anborr <73635642+anborr@users.noreply.github.com>

larswik committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcFillElementType/Documentation.md

Co-authored-by: anborr <73635642+anborr@users.noreply.github.com>

larswik committed:
PredefinedType as mandatory and added documentation for PredefinedType.

SergejMuhic committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcFillElementType/DocEntity.xml

SergejMuhic committed:
Adding documentation to IfcFillElement.PredefinedType

SergejMuhic committed:
slight correction to IfcFillElement.PredefinedType documentation

SergejMuhic committed:
Merge pull request #434 from larswik/Feaure/IfcFillElement

Add IfcFillElement/IfcFillElementType

SergejMuhic committed:
Merge pull request #428 from larswik/Feature/IfcAudioVisualAppliance-PDTs

Added enumeration constants SIREN and BEACON to IfcAudioVisualApplianceTypeEnum

SergejMuhic committed:
Merge pull request #431 from larswik/Feature/New/modified-PDTs-for-IfcDistributionSystem

PDT changes for IfcDistributionSystemEnum

larswik committed:
Added optional attribute Unit of type IfcUnit

larswik committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcVectorVoxelData/DocEntity.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

SergejMuhic committed:
Merge pull request #458 from larswik/Feature/IfcVoxelData

Adding IfcVoxelData and subclasses

SergejMuhic committed:
Merge pull request #329 from SergejMuhic/feature/voxelGrid

Adding IfcVoxelGrid

SergejMuhic committed:
Merge pull request #430 from larswik/Feature/IfcEarthingElement

Adding IfcEarthingElement/IfcEarthingElementType as new subtypes of IfcFlowTerminal/IfcFlowTerminalType

SergejMuhic committed:
Merge branch 'tunnel' into Feature/New-PDTs-for-IfcBuiltSystem

SergejMuhic committed:
Merge pull request #432 from larswik/Feature/New-PDTs-for-IfcBuiltSystem

New PDTs for IfcBuiltSystem

larswik reviewed (approved):
UNSET was fine!
Pset applicabilities handled in another PR.

SergejMuhic committed:
Merge branch 'tunnel' into feature/typeToOccurdrivenFix

SergejMuhic committed:
Revert "Merge branch 'tunnel' into feature/typeToOccurdrivenFix"

This reverts commit ab42523256c5221cf4fce48d4a27f466e0f97a52, reversing
changes made to d7e5df8058ebac39db5ef19e4308727aaef869cd.

SergejMuhic committed:
Merge branch 'develop' into feature/typeToOccurdrivenFix
